src/Take.Blip.Builder/InputExpirationHandler.cs

```diff
@@ -134,18 +142,23 @@ private bool IsMessageFromExpiration(Message message)
         /// </summary>
         /// <param name="state"></param>
         /// <param name="message"></param>
+        /// <param name="flow"></param>
         /// <returns></returns>
-        public bool IsValidateState(State state, Message message)
+        public bool IsValidateState(State state, Message message, Flow flow)
         {
             string stateToGo = string.Empty;
+            string currentSeessionState = string.Empty;
 
             return message?.Metadata?.TryGetValue(STATE_ID, out stateToGo) != true ||
                             string.IsNullOrEmpty(stateToGo) ||
                             stateToGo == IS_INPUT_EXPIRATION_FROM_SUBFLOW_REDIRECT ||
-                            state?.Id == stateToGo;
+                            (state?.Id == stateToGo) && message?.Metadata?.TryGetValue(CURRENT_SESSION_STATE, out currentSeessionState) != true ||
```

Comment:
group "(state?.Id == stateToGo) && !message?.Metadata?.TryGetValue(CURRENT_SESSION_STATE, out currentSeessionState)" in parentheses...

parentheses ((state?.Id == stateToGo) && !message?.Metadata?.TryGetValue(CURRENT_SESSION_STATE, out currentSeessionState))